client/client.go

```diff
@@ -431,12 +431,17 @@ type MetricsTimer = metrics.Timer
 // MetricsNopHandler is a noop handler that does nothing with the metrics.
 var MetricsNopHandler = metrics.NopHandler
 
-// NewClient creates an instance of a workflow client
+// NewClient creates an instance of a workflow client. This will attempt to
+// connect to the server eagerly and will return an error if the server is not
+// available.
 func NewClient(options Options) (Client, error) {
 	return internal.NewClient(options)
 }
 
-// NewNamespaceClient creates an instance of a namespace client, to manage lifecycle of namespaces.
+// NewNamespaceClient creates an instance of a namespace client, to manage
+// lifecycle of namespaces. This will not attempt to connect to the server
+// eagerly and therefore may not fail for an unreachable server until a call is
+// made.
```

Comment:
Maybe explain that you can use `WithBlock` to eagerly connect? Is that even possible?

Reply:
It is possible since we exposed gRPC options in #651. I will add the comment next commit (or right before merge if there are no reasons for another commit).

Reply:
Done